service/lib/ops/teamOps.js

```diff
@@ -332,71 +393,51 @@ var teamOps = {
   },
 
   /**
-   * Fetch specific team data
+   * Fetch the users data from a team
    *
-   * @param  {params}   params { id, organizationId }
+   * @param  {params}   params { id, page, limit }
    * @param  {Function} cb
    */
-  readTeam: function readTeam ({ id, organizationId }, cb) {
-    const tasks = []
-    let team
-
-    tasks.push((next) => {
-      const sql = SQL`
-        SELECT *
-        FROM teams
-        WHERE id = ${id}
-        AND org_id = ${organizationId}
-      `
-
-      db.query(sql, (err, result) => {
-        if (err) return next(Boom.badImplementation(err))
-        if (result.rowCount === 0) return next(Boom.notFound(`Team with id ${id} could not be found`))
+  readTeamUsers: function readTeamUsers ({ id, page, limit }, cb) {
+    const offset = (page < 1 ? 1 : page) * limit - limit
+    const job = {
+      id: id,
+      offset: offset,
+      limit: limit < 0 ? 0 : limit,
+      team: {}
+    }
 
-        team = mapping.team(result.rows[0])
+    loadTeamUsers(job, (err) => {
+      if (err) return cb(err)
 
-        team.users = []
-        team.policies = []
-        next()
-      })
+      return cb(null, job.team.users)
     })
+  },
 
-    tasks.push((next) => {
-      const sql = SQL`
-        SELECT users.id, users.name
-        FROM team_members mem, users
-        WHERE mem.team_id = ${id}
-        AND mem.user_id = users.id
-        ORDER BY UPPER(users.name)
-      `
-      db.query(sql, function (err, result) {
-        if (err) return next(Boom.badImplementation(err))
-
-        team.users = result.rows.map(mapping.user.simple)
-        next()
-      })
-    })
+  /**
+   * Fetch specific team data
+   *
+   * @param  {params}   params { id, organizationId }
+   * @param  {Function} cb
+   */
+  readTeam: function readTeam ({ id, organizationId }, cb) {
+    const job = {
+      team: {}
+    }
 
-    tasks.push((next) => {
-      const sql = SQL`
-        SELECT pol.id, pol.name, pol.version
-        FROM team_policies tpol, policies pol
-        WHERE tpol.team_id = ${id}
-        AND tpol.policy_id = pol.id
-        ORDER BY UPPER(pol.name)
-      `
-      db.query(sql, function (err, result) {
-        if (err) return next(Boom.badImplementation(err))
-
-        team.policies = result.rows.map(mapping.policy.simple)
+    async.applyEachSeries([
+      (job, next) => {
+        job.id = id
+        job.organizationId = organizationId
         next()
-      })
-    })
-
-    async.series(tasks, (err) => {
+      },
+      loadTeams,
+      loadTeamUsers,
```

Comment:
Good point to check. Probably should be made as another task as I touched this code area only to reuse some code.